sktime/forecasting/model_selection/_tune.py

```diff
@@ -182,9 +182,21 @@ def _fit(self, y, X, fh):
         self : returns an instance of self.
         """
         cv = check_cv(self.cv)
-
-        scoring = check_scoring(self.scoring, obj=self)
-        scoring_name = f"test_{scoring.name}"
+        
+        if isinstance(self.scoring, list):
+            scoring = []
+            scoring_name = []
+            for metric in self.scoring:
+                k = check_scoring(metric, obj=self)
+                kname = k.name
+                if k.name == 'DynamicForecastingErrorMetric' and hasattr(k, '__name__'):
```

Comment:
Hey can you give an example when this is happening with some code?

I tried this and it worked for me:

```python
from sktime.performance_metrics.forecasting import MeanSquaredError
m = MeanSquaredError()
m.name
# outputs 'MeanSquaredError'
```

Reply:
![Screenshot 2024-02-29 at 01 24 04](https://github.com/sktime/sktime/assets/132339317/a8a2d045-a647-466c-abdd-d65b60fff1f7)
For function metrics in sktime.performance_metrics.forecasting._functions.py for instance, this is what happens, which will lead to naming errors later on if . __ name __ is not used

Reply:
damn, do you think would it make sense to refactor `check_scoring` instead?